pandas/core/indexing.py

```diff
@@ -2417,12 +2417,53 @@ def maybe_convert_indices(indices, n):
     mask = indices < 0
     if mask.any():
         indices[mask] += n
+
     mask = (indices >= n) | (indices < 0)
     if mask.any():
         raise IndexError("indices are out-of-bounds")
     return indices
 
 
+def validate_indices(indices, n):
```

Comment:
There is similar functionality in this file (`_iLocIndexer._is_valid_integer/_is_valid_list_like`), but since that is for checking without the -1 specific behaviour, I don't think it is possible / easy to share the implementation (it would be two separate cases inside the function anyhow)

Reply:
why is this needed, as opposed to maybe_convert_indices? (maybe refactor that a bit?)

Reply:
`maybe_convert_indices` *converts* indices, this one only checks bounds. For this use-case, it is not needed to convert negative into positive indices, as this is handles well by numpy (it's only when you want to use our own `take_1d` that you need to do such conversion)

Reply:
then fix maybe_convert_indixes which duplicates this logic

Reply:
As I said, `maybe_convert_indices` *converts* the indices first and then checks bounds 0-1, this function only checks bounds but different bounds (-1 to n), so there is no real duplicate code. Combining both will only make the single function more complex to understand. Both serve a different purpose, and are used in different places in the code base. There is IMO really no reason to combine them

Reply:
FWIW, I initially started by changing `maybe_convert_indices` to handle both. But it required two new keywords to toggle checking for invalid negative indexers and whether to allow an all `-1` take from an empty index. The only thing it ended up sharing was the integer too large check, which wasn't worthwhile.

Reply:
so now we have 2 different code paths which do very similar type of validation, I would try to simplify ``Series._take`` and ``NDFrame._take`` to use these new helper functions, removing the need for ``maybe_convert_indices`` entirely.

Reply:
They have different behaviors.

1. `maybe_convert_indices` allows for any negative numbers, while `validate_indices` allows for just `-1`
2. `validate_indices` allows for an array of `-1` taken from an empty array, while `maybe_convert_indices` doesn't.

You'd need two keywords to control those behaviors, and the only then that could be shared is a `max_idx > n`. Better to have two simpler functions here.